py_experimenter/experimenter.py

```diff
@@ -110,73 +153,132 @@ def _valid_configuration(_config: configparser, credential_path=None) -> bool:
+        based on the information from the experiment configuration file the `PyExperimenter` has been initialized. 
+
+        Afterwards, the database table is filled with the list of `rows`. Note that only those rows are added which 
+        parameter combinations are not already existing. For each added row the status will be set to 'created'. 
+        If any parameter of `rows` does not match the keyfields from the experiment configuration, an error is raised. 
+        
+        :param rows: A list of rows, where each entry is made up of a dict containing a key-value-pair for each `keyfield`
+            of the experiment configuration file.
+        :type rows: List[dict]
+        :raises ValueError: If any key of any row in `rows` does not match the `keyfields` from the experiment 
+            configuration file
+        """
+        self.dbconnector.create_table_if_not_existing()
+        keyfield_names = utils.get_keyfield_names(self.config)
         for row in rows:
             if set(keyfield_names) != set(row.keys()):
                 raise ValueError('The keyfields in the config file do not match the keyfields in the rows')
-        self._dbconnector.fill_table(fixed_parameter_combinations=rows)
+        self.dbconnector.fill_table(fixed_parameter_combinations=rows)
 
-    def execute(self, approach, max_experiments: int = -1, random_order=False) -> None:
+    def execute(self, approach: Callable[[dict, dict, ResultProcessor], None], max_experiments: int = -1, random_order=False) -> None:
         """
-        Execute approach for max_experiment parameter combinations that are in the database and have status 'created'.
-        If the execution was successful, the status is changed to done and the results are written into the database. 
-        If an errors that occurs during execution the status is changed to error and the error is written into the database.
-
-        :param approach: Function to execute an experiment.
-        :param max_experiments: Max number of experiments to execute. If max_experiments is -1, all experiments are executed.
-        :param random_order:  Execute experiments in a random order.
+        Pulls open experiments from the database table and executes them.
+        
+        First the keyfield values of as many open experiments as given via `max_experiments` are pulled from the 
+        database table. An experiment is considered to be open if its status is 'created'. In case of `random_order`, 
+        they are not selected based on their consecutive experiment ID, but rather chosen randomly. Therefore the 
+        parallel pull of the same experiment ID from two instantiations of the `PyExperimenter` is avoided. 
+        
+        Afterwards, the given `approach` is executed for each set of keyfield values, changing its status to `running`.
+        Results can be written to the database table all the time via the `ResultProcessor` that is given as parameter
```

Comment:
Results can be continuously written to the database during the execution via the ...